etapas/mapas/documentacion/man/man3/cliente_lib.3
.TH "cliente_lib" 3 "Martes, 7 de Agosto de 2012" "Version 0.1" "Generacion de Mapas" \" -*- nroff -*-
.ad l
.nh
.SH NAME
cliente_lib \- 
.PP
libreria para realizar el cliente  

.SH SYNOPSIS
.br
.PP
.SS "Clases"

.in +1c
.ti -1c
.RI "class \fBcliente_lib\fP"
.br
.RI "\fIes la clase encargada del cliente \fP"
.in -1c
.SS "Funciones"

.in +1c
.ti -1c
.RI "def \fBmain\fP"
.br
.RI "\fISirve para realizar pruebas de conexion. \fP"
.in -1c
.SH "Descripción detallada"
.PP 
libreria para realizar el cliente 

Se debe especificar cual es el IP del servidor 
.PP
\fBAutores:\fP
.RS 4
Danny Vasconez 
.PP
Daniel Granda 
.RE
.PP
\fBVersión:\fP
.RS 4
0.0.3 
.RE
.PP
\fBFecha:\fP
.RS 4
2012 
.RE
.PP
\fBPrecondición:\fP
.RS 4
Tener funcionando el servidor 
.RE
.PP
\fBBug\fP
.RS 4
myTemperature tiene falla por el tipo de dato en formato a python '' toca pasar a '0x81', devuelve_valors fallo cuando no tiene valores 
.RE
.PP
\fBAtención:\fP
.RS 4
uso inapropiado puede hacer que la aplicacion falle
.RE
.PP
.SH "Ejemplo de uso"
.PP
En el ejemplo se muestra tres maneras de enviar comandos la general que es requestOnce y las otras que son la misma pero modificada para trabajar con comandos especificos 
.PP
.nf
a=cliente_lib() #Instancia a la clase cliete_lib
a.ip="192.168.1.10"
CLIENTE=a.cliente_inicio() 
# para realizar movimiento 
TransRatio,RotRatio,LatRatio = [-50,0,0]
CLIENTE=a.envio_ratioDrive(CLIENTE,TransRatio,RotRatio,LatRatio) #fijar los valores para mover
#
#para conocer valores como fisicos de la plataforma movil
CLIENTE=a.envio_consulta_fisica(CLIENTE,"updateNumbers")
valor=a.devuelve_valorf()
print valor
#
#para conocer los valores de los sonares
CLIENTE.requestOnce("pose")
valor=a.devuelve_valors()
print valor
#
ArUtil.sleep(1000)
a.cliente_apaga(CLIENTE)


.fi
.PP
 
.SH "Documentación de las funciones"
.PP 
.SS "def cliente_lib.main ()"
.PP
Sirve para realizar pruebas de conexion. sin tener que ejecutar la aplicacion completa; de la siguiente forma 'python2.5 cliente_lib.py' 
.PP
\fBDevuelve:\fP
.RS 4
0 
.RE
.PP

.PP
Definición en la línea 318 del archivo cliente_lib.py.
.PP
.nf
319           :
320         #prueba de la libreria
321         a=cliente_lib()
322         #a.ip='192.168.10.5' #Si el servidor esta en otra maquina 
323         CLIENTE=a.cliente_inicio()
324         #TransRatio,RotRatio,LatRatio = [-50,0,0]
325         #CLIENTE=a.envio_ratioDrive(CLIENTE,TransRatio,RotRatio,LatRatio) #fijar los valores para mover
326         TransRatio=5
327         #for i in range(500):
328                 #CLIENTE=a.envio_consulta_fisica(CLIENTE,'pose')
329         CLIENTE=a.envio_griper(CLIENTE,TransRatio) #fijar los valores para mover
330         ArUtil.sleep(500)
331                 #print 'Longitud X',len(a.x)
332                 #print 'Longitud Y',len(a.y)
333         #print 'X:', a.x
334         #print 'Y:', a.y
335                 
336         #CLIENTE.requestOnce('pose')
337         #ArUtil.sleep(100)
338         a.cliente_apaga(CLIENTE)
        return 0
.fi
.SH "Autor"
.PP 
Generado automáticamente por Doxygen para Generacion de Mapas del código fuente.


CHANGELOG
Version 0.1.6b 2011-xx-xx
------------------------
2 modulos xml para los temas (themes/<tema>/*.xml) para saber cuales
  son las etiquetas y valores a usar 
  <<si consigo esto sube a 0.1.7a-RC1 >>
  <<si consigo esto sube a 0.1.7a pero consiguiendo punto 3 y 2>>
3 modulos xml para enlistar los temas la etiqueta tema contiene el 
  nombre del arvhico svg (ej reloj/protector.svg) y la etiqueta el 
  nombre del tema "wallv2" quedando de esta manera
  <theme nombre="wallv2">reloj/protector.svg</theme>  
  <<<si consigo esto sube a 0.1.7a-RC1 >>>
  <<<si consigo esto sube a 0.1.7a pero consiguiendo punto 2 y 3>>>
<4 conseguir cambiar el fondo de pantalla y su GUI
  <<<si consigo esto subo a 0.1.6a>>>
  <<<si consigo esto sube a 0.1.7a-RC1  si consegi punto 2>>>
  <<<si consigo esto sube a 0.1.7a-RC1  si consegi punto 3>>>
  <<<si consigo esto sube a 0.1.7b  si consegi punto 2 y 3>>>
<5 Consiguir widget-pasivos en la libreria Ergus 
<6 Consiguir widget-activos en la libreria Ergus 
<7 Consiguir widget-interactivos en la libreria Ergus
8 Construir la GUI del clic izq que es para configuracion general y todos los puntos anteriores
  <<Sube a la version 0.2a>> 

Version 0.1.6b 2011-xx-xx
- Icarus sube a la version 0.1b integrado con icarus_fondo
- Demonios wallv3 y poweroff utilizan los archivos "protector.xml" de cada uno para poder saber informacion de dimensiones y posiciones del widgets
- icarus_panel version 0.1_alfa. Sirve como panel para el escritorio por el momento solo tiene un windows_selector

Version 0.1.6a 2011-07-15
- Corregido bugs instalador y del GUI
- Autostart multiple en icarus-gui en el archivo "demonio.xml"
- Anadido widget en modo pasivo, activo y interactivo a Ergus
- Icarus_demonio_wall sube a la version 0.3
- icarus_pantallazo sube a la version 0.2
- icarus_demonio_poweroff version 0.1 
- icarus_fondo version 0.1_alfa. Sirve como wallpaper-setter basado en csetroot de circus
- Anadido archivo "pantallazo.xml" para saber que opciones tiene pantallazo
- Modificado iconstatus de icarus-gui a
   *Anadir fondo de pantalla (icarus_fondo v0.1-alfa)
   *Seleccionar demonio (integra a "activar/desactivar demonio")
   *Ponar captura de pantalla (icarus_pantallazo v0.2)
   *Acerca de
   *Salir

Version 0.1.5, 2011-06-23
- anadido iconstatus para el tray del sistema con las opciones:
   *Activar demonio
   *Desactivar demonio
   *Seleccionar demonio
   *Ponar captura de pantalla (icarus_pantallazo v0.1)
   *Acerca de
   *Salir
- Anadido archivo "demonio.xml" para saber que demonios se tiene para utilizar  
- Icaurs_demonio_wall sube a la version 0.2

Version 0.1, 2011-06-07
----------------------
- "First" release.


etapas/teleoperacion/cliente_lib.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#       cliente_lib.py
#       
#       Copyright 2012 Danny Enrique Vasconez <dannyvasconeze@gmail.com>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.

from AriaPy import *
from ArNetworkingPy import *
import sys
"""\package cliente_lib
   \brief libreria para realizar el cliente
   \details Se debe especificar cual es el IP del servidor
   \authors    Danny Vasconez
   \authors   Daniel Granda
   \version   0.0.1
   \date      2012
   \pre       Tener funcionando el servidor
   \bug       Nada
   \warning   uso inapropiado puede hacer que la aplicacion falle
   
"""
"""
   \section intro Ejemplo de uso
   En el ejemplo se muestra tres maneras de enviar comandos la general que es requestOnce y las otras que son la misma pero modificada para trabajar con comandos especificos
   \verbinclude ejemplo_cliente
"""

def main():
	
	return 0

def valores(packet):
	"""
	\brief Sirve cuando se manda el comando "test" en el paquete
	\param packet el paquete que recibe el cliente del servidor 
	\return Nada, pero imprime los valores X,Y,Th en el terminal
	"""
	voltaje_vateria=packet.bufToByte2()/10
	myX = packet.bufToByte4()#
	myY = packet.bufToByte4()
	myTh = packet.bufToByte2()
	myVel = packet.bufToByte2()
	myRotVel = packet.bufToByte2()
	myLatVel = packet.bufToByte2()
	myTemperature = packet.bufToByte()
	print "X= "+str(myX)+" y="+str(myY)+" th="+str(myTh)

def envio_ratioDrive(client,TransRatio,RotRatio,LatRatio):
	"""
	\brief Sirve para realizar la teleoperacion, mandando los parametros 
	\param client Se debe trar el objeto cliente a la definicion para poder utilizar el enlace del cliente para enviar el paquete al servidor
	\param TransRatio Velocidad de traslacion
	\param RotRatio Velocidad de rotacion
	\param LatRatio velocidad lateral para el modelo Pioneer P3-DX no se necesario puede ser 0
	\return client
	"""
	myTransRatio=TransRatio
	myRotRatio=RotRatio
	myLatRatio=LatRatio
	packet=ArNetPacket()
	packet.doubleToBuf(myTransRatio)
	packet.doubleToBuf(myRotRatio)
	packet.doubleToBuf(50) # use half of the robot's maximum.
	packet.doubleToBuf(myLatRatio)
	client.requestOnce("ratioDrive", packet)
	return client
  
def uC_comandos_movi(client,comando,parametro):
	"""
	\brief Sirve para mandar ordenes de movimiento directamente al controlador de la plataforma movil 
	\param client Se debe trar el objeto cliente a la definicion para poder utilizar el enlace del cliente para enviar el paquete al servidor
	\param comando es un numero de 1-255 que representa una funcion esta informacion se puede encontrar en el API de ARIA
	\param parametro el parametro de la funcion en caso de no tener se deja el valor en blanco
	\return client
	"""
	mi_comando=comando #comando 8 es MOVE parametro un valor de 5000 a -4999 es en mm, 11 LEV y su parametro es velocidad +o- mm/s
	mi_parametro=parametro #parametro
	packet=ArNetPacket()
	packet.strToBuf(mi_comando+" "+mi_parametro)
	client.requestOnce("MicroControllerMotionCommand", packet) #MicroControllerMotionCommand
	return client
	
def cliente_inicio(ip="localhost"):
	"""
	\brief Sirve para iniciar la conexion con el servidor
	\param ip="localhost" El valor de local host se lo debe cambiar por el IP del servidor
	\return client
	"""
	client = ArClientBase()
	Aria.init()
	
	startTime = ArTime()
	startTime.setToNow()
	if not client.blockingConnect(ip, 7272): #ip y puerto del servidor
		print "Could not connect to server at localhost port 7272, exiting"
		Aria.exit(1);
	print "cliente: Se tardo %ld msec en connectarse\n" % (startTime.mSecSince())
	
	client.runAsync()
	client.addHandler("updateNumbers",valores) #Verifica y agrega a la cabecera de comandos para el servidor
	if client.dataExists("ratioDrive"): #comprobar si existe el comando ratioDrive
		print "ratioDrive si existe"
	else:
		Aria.exit(1);
	#client=envio_ratioDrive(client,TransRatio,RotRatio,LatRatio) #fijar los valores para mover
	#client=uC_comandos_movi(client,comando,parametro) #Lo hace de una manera directa anulando las demas operaciones
	#client.requestOnce("updateNumbers")
	#client.requestOnce("stop") #parada de emergencia
	return client
def cliente_apaga(client):
	"""
	\brief Sirve para realizar la desconexion con el servidor
	\return Nada
	"""
	ArUtil.sleep(1000)
	client.disconnect()
	ArUtil.sleep(50)
	return 0

class cliente_lib:
	"""
	\class cliente_lib.cliente_lib
	\brief No dispone ninguna utilidad.
	\details  Solo se puso para mantener la estructura de python
	"""
	def __init__(self):
		print "Cargo modulo para cliente_lib"
if __name__ == '__main__':
	main()

